sidecar/supervisor/appsupervisor.go

```diff
@@ -94,8 +103,12 @@ func (a *AppSupervisor) DoAppSupervision(cmdArgs []string, agent *register.Regis
 func (a *AppSupervisor) Shutdown(exitCode int) {
 	// TODO: Gracefully shutdown Ngnix, and filebeat
 	//ugly temporary hack to kill all processes in container
-	exec.Command("pkill", "-9", "-f", "nginx")
-	exec.Command("pkill", "-9", "-f", "filebeat")
+	//exec.Command("pkill", "-9", "-f", "nginx")
+	//exec.Command("pkill", "-9", "-f", "filebeat")
+
+	for _, cmd := range a.cmds {
+		cmd.Process.Kill()
+	}
 
 	if a.agent != nil {
 		a.agent.Stop()
```

Comment:
There seem to be a bug here (from #347). After the agent has been stopped, it might be immediately restarted by the healthchecker. An optional solution is to not pass in `RegistrationAgent`, but rather pass in some `Startable` interface, which is implemented by both the `RegistrationAgent` and healthchecker.